lib/hammer_cli/output/output.rb

```diff
@@ -69,5 +71,11 @@ def init_adapter(adapter_name)
       @adapter ||= self.class.adapters[adapter_name].new(context, self.class.formatters)
     end
 
+    def adjust_definition(definition)
+      return unless @context.key?(:only)
+      return if adapter.class <= HammerCLI::Output::Adapter::TreeStructure
+      labels = @context[:only].map(&:downcase)
+      definition.select_fields!(deep_select: true, labels: labels)
```

Comment:
Using `select_fields!` will remove the fields from the command class for good. This will disallow using the option repeatedly in shell mode:
```
hammer shell
> host list --fields Id,Name
... will list the hosts
> host list --fields Id,Name,Subnet
... will break
```
Note that at the moment it's not possible to use the option in shell mode at all (see my previous comment).